src/python/pants/rules/core/list_roots.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.console import Console
+from pants.engine.goal import Goal, LineOriented
+from pants.engine.rules import console_rule, optionable_rule
+from pants.source.source_root import SourceRootConfig
+
+
+class Roots(LineOriented, Goal):
+  """List the repo's registered source roots."""
+  name = 'roots'
+
+
+@console_rule(Roots, [Console, Roots.Options, SourceRootConfig])
+def list_roots(console, options, source_root_config):
+  all_roots = source_root_config.get_source_roots().all_roots()
```

Comment:
@benjyw : See my previous comment on this phab. This line escapes the sandbox, and will break under pantsd.

Reply:
Had to remind myself how this code works, but yes, this needs fixing. I couldn't figure out how it was working today, but it turns out that this is the only thing that uses `all_roots()`. Thanks for the heads up.